src/server/req_preemptjob.c

```diff
@@ -85,34 +277,73 @@ extern time_t time_now;
 void
 req_preemptjobs(struct batch_request *preq)
 {
-	int			i = 0;
-	int			count = 0;
-	job			*pjob = NULL;
-	preempt_job_info	*ppj = NULL;
-	pbs_sched		*psched;
+	int i = 0;
+	int count = 0;
+	job *pjob = NULL;
+	preempt_job_info *ppj = NULL;
+	pbs_sched *psched;
 
 	preq->rq_reply.brp_code = 0;
 	count = preq->rq_ind.rq_preempt.count;
 	psched = find_sched_from_sock(preq->rq_conn);
 
+	preempt_total = preq->rq_ind.rq_preempt.count;
+
+	if (preempt_jobs_list == NULL) {
+		if ((preempt_jobs_list = calloc(sizeof(preempt_job_info), preempt_total)) == NULL) {
+			req_reject(PBSE_SYSTEM, 0, preq);
+			log_err(errno, __func__, "Unable to allocate memory");
+			return;
+		}
+	}
+
+	preq->rq_reply.brp_choice = BATCH_REPLY_CHOICE_PreemptJobs;
+	preq->rq_reply.brp_un.brp_preempt_jobs.count = preq->rq_ind.rq_preempt.count;
+
 	for (i = 0; i < count; i++) {
 		ppj = &(preq->rq_ind.rq_preempt.ppj_list[i]);
 		pjob = find_job(ppj->job_id);
+		/* The job is already out of the way.  This must of happened after the scheduler
```

Comment:
Well this comment led me down an interesting rabbit hole.  20+ years ago when I was taught to type, two spaces was the norm after a period.  I talked to @agurban and she said that one or two is fine and she uses two.  She also pointed me at the Chicago Manual of Style as the definitive place to look.  They said the CMoS says one space.  Life changes when you move from a monospaced typewriter to a computer.  I guess I have old habits and will have to start changing.  Since most people who work on PBS started typing on a computer, they will use one space.  I'll be consistent and change it.